src/app/wallet/proposals/add-proposal/add-proposal.component.ts

```diff
@@ -0,0 +1,151 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { Log } from 'ng2-logger';
+import { Observable } from 'rxjs/Observable';
+
+import { FormArray } from '@angular/forms/src/model';
+import { ProposalsService } from 'app/wallet/proposals/proposals.service';
+import { ProfileService } from 'app/core/market/api/profile/profile.service';
+import { SnackbarService } from 'app/core/snackbar/snackbar.service';
+import { Profile } from 'app/core/market/api/profile/profile.model';
+import { PeerService } from 'app/core/rpc/peer/peer.service';
+
+import {
+  ProposalConfirmationComponent
+} from 'app/modals/proposal-confirmation/proposal-confirmation.component';
+import { ModalsHelperService } from 'app/modals/modals-helper.service';
+import { Proposal } from 'app/wallet/proposals/models/proposal.model';
+
+@Component({
+  selector: 'app-add-proposal',
+  templateUrl: './add-proposal.component.html',
+  styleUrls: ['./add-proposal.component.scss']
+})
+export class AddProposalComponent implements OnInit {
+
+  log: any = Log.create('add-item.component');
+  private destroyed: boolean = false;
+  public isTnCAccepted: boolean = false;
+  private profileId: number;
+  // form controls
+  public proposalFormGroup: FormGroup;
+
+  constructor(
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private dialog: MatDialog,
+    private modalService: ModalsHelperService,
+    private peerService: PeerService,
+    private proposalsService: ProposalsService,
+    private profileService: ProfileService,
+    private snackbarService: SnackbarService
+  ) { }
+
+  ngOnInit() {
+    // get default profile address.
+    this.profileService.default().takeWhile(() => true).subscribe((profile: Profile) => {
```

Comment:
It seems that the common idiom used in the code is `.takeWhile(() => !this.destroyed)` but weare using `.takeWhile(() => true)`. Which would never complete until the app ends.

Reply:
Yeah, I missed it, thanks @zaSmilingIdiot.